code/game/machinery/limbgrower.dm

```diff
@@ -0,0 +1,256 @@
+
+	if(user.a_intent == "harm") //so we can hit the machine
+		return ..()
+
+	if(stat)
+		return
+
+	if(O.flags & HOLOGRAM)
+		return
+
+/obj/machinery/limbgrower/Topic(href, href_list)
+	if(..())
+		return
+	if (!busy)
+		if(href_list["menu"])
+			screen = text2num(href_list["menu"])
+
+		if(href_list["category"])
+			selected_category = href_list["category"]
+
+		if(href_list["disposeI"])  //Get rid of a reagent incase you add the wrong one by mistake
+			reagents.del_reagent(href_list["disposeI"])
+
+		if(href_list["make"])
+
+			/////////////////
+			//href protection
+			being_built = files.FindDesignByID(href_list["make"]) //check if it's a valid design
+			if(!being_built)
+				return
+
+
+			var/synth_cost = being_built.reagents["synthflesh"]*prod_coeff
+			var/power = max(2000, synth_cost/5)
+
+			if(reagents.has_reagent("synthflesh", being_built.reagents["synthflesh"]*prod_coeff))
+				busy = 1
+				use_power(power)
+				flick("limbgrower_fill",src)
+				icon_state = "limbgrower_idleon"
+				addtimer(src, "build_item",32*prod_coeff)
+
+	else
+		usr << "<span class=\"alert\">The limb grower is busy. Please wait for completion of previous operation.</span>"
+
+	updateUsrDialog()
+	return
+
+/obj/machinery/limbgrower/proc/build_item()
+	if(reagents.has_reagent("synthflesh", being_built.reagents["synthflesh"]*prod_coeff))	//sanity check, if this happens we are in big trouble
+		reagents.remove_reagent("synthflesh",being_built.reagents["synthflesh"]*prod_coeff)
+		var/B = being_built.build_path
+		if(ispath(B, /obj/item/bodypart))	//This feels like spatgheti code, but i need to initilise a limb somehow
+			build_limb(B)
+		else
+			//Just build whatever it is
+			new B(loc)
+	else
+		usr << "<span class=\"error\"> Something went very wrong and there isnt enough synthflesh anymore!</span>"
+	busy = 0
+	flick("limbgrower_unfill",src)
+	icon_state = "limbgrower_idleoff"
+	updateUsrDialog()
+
+/obj/machinery/limbgrower/proc/build_limb(var/B)
+	//i need to create a body part manually using a set icon (otherwise it doesnt appear)
+	var/obj/item/bodypart/L
+	var/selected_part
+	L = new B(loc)
+	if(selected_category=="human" || selected_category=="lizard")
+		L.icon = 'icons/mob/human_parts_greyscale.dmi'
+	// gotta check which part of the body it is, so that i can get the correct icon
+	if(ispath(B,/obj/item/bodypart/r_arm))
+		L.icon_state = "[selected_category]_r_arm_s"
+		selected_part = "right arm"
+	else if(ispath(B,/obj/item/bodypart/l_arm))
+		selected_part = "left arm"
+		L.icon_state = "[selected_category]_l_arm_s"
+	else if(ispath(B,/obj/item/bodypart/r_leg))
+		selected_part = "right leg"
+		L.icon_state = "[selected_category]_r_leg_s"
+	else if(ispath(B,/obj/item/bodypart/l_leg))
+		selected_part = "left leg"
+		L.icon_state = "[selected_category]_l_leg_s"
+	L.name = "Synthetic [selected_category] Limb"
+	L.desc = "A synthetic [selected_category] limb that will morph on its first use in surgery. This one is for the [selected_part]"
```

Comment:
I've tested them out and they attach fine to mobs. They are just regular arms and legs with the "dropped" value set to false. I set them a icon because otherwise they would be invisible (but still work) as they havent been "dropped" from a mob so dont know what they should look like.